(categorize): update medical imaging

alxndrkalinin opened the pull request:
This is work in progress to update `Categorize` section with challenges and solutions in medical imaging.

Todo list:
- [x] CAD methods in radiography, CT, MRI (e.g. #336)
- [x] skin cancer papers (e.g. #207)
- [x] DR and AMD (e.g. #122, #151)

Not ready for review yet, but should be soon.

enricoferrero commented on sections/03_categorize.md:
```diff
@@ -47,53 +47,67 @@ regards to data availability and labeling.
+machine learning techniques, for example, as a cascade of deep learning and
+random forest models [@tag:Dhungel2017_mamm_min_interv]. Using larger dataset
+of around 45,000 images, Kooi et al. [@tag:Kooi2016_mamm_lesions] demonstrated
+that deep neural networks can outperform the traditional CAD system at low
```
Maybe say what CAD stands for?

cgreene replied:
It only gets used once from what I see - I'd say expand out the acronym at least, and define if it's valuable for readers to know the acronym as well.

alxndrkalinin replied:
done

enricoferrero commented on sections/03_categorize.md:
```diff
@@ -47,53 +47,67 @@ regards to data availability and labeling.
+incorporate into existing tools to aid pathologists. The authors' work suggests
+that this could reduce the false negative rate of such evaluations. This theme
+of an ensemble between deep learning algorithm and human expert may help
+overcome some of the challenges presented by data limitations.
```
@agitter @cgreene I think this is an important point and might be worth emphasising more, maybe in the Discussion or Conclusions.

We touch on this theme a couple of times in Categorise, e.g.:
> We don't expect that deep learning methods will replace expert review. We expect them to complement expert review by allowing more efficient use of the costly practice of manual annotation.

But maybe we should be more explicit about it?

cgreene replied:
I agree - @enricoferrero : can you create an issue & address via your own PR when this PR goes through?

agitter replied:
Agreed.  I touched on this in the conclusion

> In medical imaging, deep learning can point an expert to the most challenging cases that require manual review [@doi:10.1148/radiol.2017162326], though the risk of false negatives must be addressed.

but this could be expanded and emphasized.

enricoferrero commented on sections/03_categorize.md:
```diff
@@ -47,53 +47,67 @@ regards to data availability and labeling.
-when training had concluded. In other domains, this has resulted in a minimal
-change in the estimated accuracy
-[@url:http://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf].
+health record. Recently Lee et al.[@tag:Lee2016_emr_oct_amd] developed an
```
> are electronic health records

instead of
> is the electronic health record

?

alxndrkalinin replied:
fixed

enricoferrero commented on sections/03_categorize.md:
```diff
@@ -47,53 +47,67 @@ regards to data availability and labeling.
-change in the estimated accuracy
-[@url:http://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf].
+health record. Recently Lee et al.[@tag:Lee2016_emr_oct_amd] developed an
+approach to distinguish individuals with Age-related Macular Degeneration
```
I would drop the capitalisation here.

alxndrkalinin replied:
fixed

enricoferrero commented on sections/03_categorize.md:
```diff
@@ -47,53 +47,67 @@ regards to data availability and labeling.
+techniques, the authors reach greater than 93% accuracy. One item that is
+important to note with regards to this work is that the authors used their test
+set for evaluating when training had concluded. In other domains, this has
+resulted in a minimal change in the estimated accuracy
```
Not convinced this caveat is worth mentioning to be honest.

alxndrkalinin replied:
It was not my contribution and it was approved before, so I don't feel convinced to remove it right away. Maybe it's worth looking at once again at the final paper reviewing stage?

alxndrkalinin commented:
@agitter @cgreene sorry for the delay, I think this is mostly done now. Next round of review is appreciated.

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
 
 One of the general areas where deep learning methods have had substantial
```
"One area where"

alxndrkalinin replied:
fixed

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+image registration, content-based image retrieval, image generation and
+enhancement, and combining image data with clinical reports. For the extensive
+literature review, including approaches and applications see recent papers
+[@tag:Litjens2017_medimage_survey @tag:Shen2017_medimg_review].
```
I think just tag these references on the end of the previous sentence and save a line.

alxndrkalinin replied:
agreed

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+Closest to natural images are applications of deep learning aimed at
+detection and recognition of melanoma, the deadliest form of skin cancer.
+Recent works included applications to both dermoscopy
+[@tag:Codella2016_ensemble_melanoma @tag:Yu2016_melanoma_resnet] and simply
```
What does "simply" refer to here?

alxndrkalinin replied:
re-phrased

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+clinical photography images of skin lesions [@tag:Jafari2016_skin_lesions
+@tag:Esfahani2016_melanoma @tag:Esteva2017_skin_cancer_nature]. For both
+modalities pre-training on natural images appears to be common model
+initialization, applied networks can be as deep as over 50 layers, and result
```
I think the "50 layers" phrase can be deleted

alxndrkalinin replied:
agreed

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+subsequent fine tuning [@tag:Dhungel2016_mamm]. Adaptation to the medical image
+domain can further improved by combining in the latter approach with other
+machine learning techniques, for example, as a cascade of deep learning and
+random forest models [@tag:Dhungel2017_mamm_min_interv]. Using larger dataset
```
"a larger"? And larger than what? Actually, the number could be excluded.

alxndrkalinin replied:
fixed

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+sensitivity. They also compared network performance to certified screening
+radiologists on a patch level and found no significant difference between the
+network and the readers. Similarly, Geras et al. [@tag:Geras2017_multiview_mamm]
+showed that both having larger dataset and using multi-view network architecture
```
maybe "using a large dataset"?

alxndrkalinin replied:
re-phrased

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+that do not have as much data using transfer learning [@tag:Kooi2017_mamm_tl].
+Though this strategy has not yet been employed in this domain, high-quality
+feature detectors can be constructed from large collections of unlabeled images
+in an unsupervised context. Then the small number of labeled examples could
```
Delete "Then"

alxndrkalinin replied:
fixed

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+scratch [@tag:Bar2015_nonmed_tl @tag:Shin2016_cad_tl
+@tag:Rajkomar2017_radiographs @tag:Lakhani2017_radiography]. All these
+studies demonstrate successful transfer of features learnt from natural image
+datasets, such as ImageNet [@tag:Russakovsky2015_imagenet]. Rajkomar et al.
```
Sentence beginning "Rajkomar" sounds funny - missing a word.

alxndrkalinin replied:
Split in 2

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+few dozen patients can outperform a pre-trained out-of-domain classifier.
+Nie et al. [@tag:Nie2016_3d_survival] showed that multimodal, multi-channel
+3D deep architecture was successful at learning high-level brain tumor
+appearance features jointly from T1 MRI, fMRI and DTI images, outperforming
```
Do we have these abbreviations defined and do we need them. Maybe just says "MRI"?

alxndrkalinin replied:
expanded, maybe worth adding to the list of abbreviations #345

agapow commented on sections/03_categorize.md:
```diff
@@ -46,65 +46,167 @@ regards to data availability and labeling.
+In more recent work, Wang et al. [@tag:Wang2016_breast_cancer] analyzed stained
+slides to identify cancers within slides of lymph node slices. The approach
+provided a probability map for each slide. On this task a pathologist has about
+a 3% error rate. The pathologist did not produce any false positives, but did
```
Maybe merge a few of these senetences to shorten length and make it read a little easier. e.g.

While the algorithm had about twice the error rate of a pathologist, the errors were not strongly correlated with those of the pathologist, suggesting that the two could be combined for a joint error rate of under 1%.

alxndrkalinin replied:
Done

alxndrkalinin commented:
@agapow thanks for the review! I addressed your points and added a summary paragraph.

I am open to further high-level re-organization suggestions as well.

cgreene commented:
My inclination is to merge this (if the CI run completes). Then @agapow or anyone else can propose additional edits. I think that in these last few days we will try to keep the PRs flowing quickly.

cgreene commented:
CI run was successful. Merging. Participants (or anyone else) should feel free to file a new PR to add more edits if desired.
